src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RegistrationException("Login and password cannot be null");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters.");
+        }
+
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters.");
+        }
+
+        if (user.getAge() < 18) {
+            throw new RegistrationException("User must be at least 18 years old.");
```

Comment:
The checklist warns against tying checks to values that may change over time, such as the maximum human lifespan. If the minimum age of 18 is not a requirement for the task, this check should be removed. If it is a requirement, the value '18' should be a named constant to avoid magic numbers.